### PR TITLE
271 vite updates

### PR DESCRIPTION
**Pull request by Vadim121197**

### CODE REVIEW

**Comment by Vadim121197** [comment]
sorry, need more time to fix  error - 
`does not provide an export named 'Buffer'`

File: apps/extension/vite.config.mts
Comment (grod220) [inline apps/extension/vite.config.mts:1]: Renaming these files to `.mts` should not be necessary

File: packages/types/src/hex.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import { Buffer } from 'buffer';
```
Comment (grod220) [inline packages/types/src/hex.ts:1]: We don't want to rely upon globals

File: packages/crypto/browser.config.mts [hunk 2]
```diff
@@ -1,7 +1,10 @@
 /// <reference types="vitest" />
 import { defineConfig } from 'vite';
+import { nodePolyfills } from 'vite-plugin-node-polyfills';
 
 export default defineConfig({
+  // fix error - Buffer is not defined
+  plugins: [nodePolyfills()],
```
Comment (grod220) [inline packages/crypto/browser.config.mts:7]: Why is this needed if we have the "buffer" package in our package.json?

File: apps/extension/vite.config.mts
Comment (Vadim121197) [reply apps/extension/vite.config.mts:1]: it fix warning - The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

Comment (Vadim121197) [reply apps/extension/vite.config.mts:1]: Знімок екрана 2023-12-05 о 18 17 22

Comment (grod220) [reply apps/extension/vite.config.mts:1]: Adding `"type": "module"` in the package.json should be fine

File: packages/crypto/browser.config.mts (hunk 2 above)
Comment (Vadim121197) [reply packages/crypto/browser.config.mts:7]: fix  `Buffer is not defined` when i start test

File: apps/extension/vite.config.mts
Comment (Vadim121197) [reply apps/extension/vite.config.mts:1]: Знімок екрана 2023-12-05 о 18 39 41

i should rename files to .cjs

File: packages/types/src/hex.ts (hunk 1 above)
Comment (grod220) [reply packages/types/src/hex.ts:1]: My overall feedback is that we need to find a better solution in general to this problem. Moving to a reliance upon globals in our production app in order to run our tests is not a good approach.

**Comment by grod220** [comment]
My overall feedback is that we need to find a better solution in general to this problem. Moving to a reliance upon globals in our production app in order to run our tests is not a good approach.

**Comment by Vadim121197** [comment]
but we do as global when use extension -  
```
 new webpack.ProvidePlugin({
      Buffer: ['buffer', 'Buffer'],
    }),
```

**Comment by grod220** [comment]
But doesn't our npm package dependency `"buffer": "^6.0.3"` remove the requirement of relying on a global?

**Comment by grod220** [comment]
Realizing that our import should look like this:

```typescript
import { Buffer } from 'buffer/';
```

This indicates that it's not the global buffer module, but the npm package that should be imported. Think this will help us get around these global issues?

**Comment by Vadim121197** [comment]
this library works - https://www.npmjs.com/package/vite-plugin-node-stdlib-browser! but it's not  safe types, I'll try their implementation! strangely, other similar libraries don't work

**Comment by Vadim121197** [comment]
can there be a problem that the library exposes the value as exports.Buffer and because of this I get the error The requested module `'web/node_modules/.pnpm/registry.npmmirror.com+buffer@6.0.3/node_modules/buffer/index.js?v=aed3b625' does not provide an export named 'Buffer'`?
